components/vfs/manager.cpp

```diff
@@ -57,6 +57,14 @@ namespace VFS
             (*it)->listResources(mIndex, mStrict ? &strict_normalize_char : &nonstrict_normalize_char);
     }
 
+    void Manager::rebuildIndex()
+    {
+        for (std::vector<Archive*>::const_iterator it = mArchives.begin(); it != mArchives.end(); ++it)
+            (*it)->reset();
+
+        buildIndex();
+    }
```

Comment:
Recreating it is probably not going to work. It requires something like
```
if (mVFS)
        *mVFS.get() = VFS::Manager(mFsStrict);
```
OR adding several setVFS functions. The most troublesome aspect of the latter is the requirement to iterate through all cells to update the pointer in ESMTerrain::Storage.